internal/cli/fmt.go

```diff
@@ -0,0 +1,142 @@
+package cli
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/posener/complete"
+
+	"github.com/hashicorp/waypoint-plugin-sdk/terminal"
+	"github.com/hashicorp/waypoint/internal/clierrors"
+	configpkg "github.com/hashicorp/waypoint/internal/config"
+	"github.com/hashicorp/waypoint/internal/pkg/flag"
+)
+
+type FmtCommand struct {
+	*baseCommand
+
+	flagWrite bool
+}
+
+func (c *FmtCommand) Run(args []string) int {
+	// Initialize. If we fail, we just exit since Init handles the UI.
+	if err := c.Init(
+		WithArgs(args),
+		WithFlags(c.Flags()),
+		WithNoConfig(),
+		WithClient(false),
+	); err != nil {
+		return 1
+	}
+
+	// If we have too many args, error immediately.
+	if len(c.args) > 1 {
+		c.ui.Output("At most one argument is expected.\n\n"+c.Help(), terminal.WithErrorStyle())
+		return 1
+	}
+
+	// If we have no args, default to the filename
+	if len(c.args) == 0 {
+		c.args = []string{configpkg.Filename}
+	}
+
+	// Read the input
+	src, err := c.readInput()
+	if err != nil {
+		c.ui.Output(
+			"Error reading input to format: %s", clierrors.Humanize(err),
+			terminal.WithErrorStyle(),
+		)
+		return 1
+	}
+
+	// Format it
+	name := "<stdin>"
+	stdin := true
+	if c.args[0] != "-" {
+		name = filepath.Base(c.args[0])
+		stdin = false
+	}
+	out, err := configpkg.Format(src, name)
+	if err != nil {
+		c.ui.Output(
+			"Error formatting: %s", clierrors.Humanize(err),
+			terminal.WithErrorStyle(),
+		)
+		return 1
+	}
+
+	// If we're writing then write it to the file. stdin never writes to a file
+	if c.flagWrite && !stdin {
+		if err := ioutil.WriteFile(c.args[0], out, 0644); err != nil {
+			c.ui.Output(
+				"Error writing formatted output: %s", clierrors.Humanize(err),
+				terminal.WithErrorStyle(),
+			)
+			return 1
+		}
+	} else {
+		// We must use fmt here and not c.ui since c.ui may wordwrap and trim.
+		fmt.Print(string(out))
+	}
+
+	return 0
+}
+
+func (c *FmtCommand) readInput() ([]byte, error) {
+	// If we have non-stdin input then read it
+	if c.args[0] != "-" {
+		return ioutil.ReadFile(c.args[0])
+	}
+
+	// Otherwise it is stdin
+	return ioutil.ReadAll(os.Stdin)
+}
+
+func (c *FmtCommand) Flags() *flag.Sets {
+	return c.flagSet(0, func(sets *flag.Sets) {
+		f := sets.NewSet("Command Options")
+
+		f.BoolVar(&flag.BoolVar{
+			Name:    "write",
+			Target:  &c.flagWrite,
+			Default: false,
+			Usage: "Overwrite the input file. If this is false, the formatted " +
```

Comment:
Quick note, this should be reversed to have the opposite behavior. Always write in place, and can be disabled and write to stdout with `-write=false`